specs/src/tx_client_v2.md

```diff
@@ -0,0 +1,148 @@
+# Tx Client v2 – Sequential Submission Queue
+
+### Context & Motivation
+
+Tx Client v2 enforces a per-account sequential submission queue that serializes signing, submission, and recovery, avoiding sequence races and minimizing transaction failures.
+
+## Protocol/Component Description
+
+### Per account sequential queue
+
+Every account gets its own **transaction queue**.
+
+This queue is in charge of everything for that account: taking new transactions(jobs), broadcasting them in submission order, reporting outcomes.
+
+Inside the queue we keep two kinds of transactions:
+
+- **Queued** - submitted but not broadcast yet
+- **Pending** - already broadcast and waiting for confirmation
+
+---
+
+### Broadcast and Confirmation loops
+
+Each queue runs two background loops:
+
+1. **A submitter**
+    - Picks the next queued tx, signs it, broadcasts it to the node.
+    - Listens for evicted transactions and resubmits them.
+2. **A monitor**
+    Checks the statuses of pending txs and knows how to manage evictions/rejections/confirmations.
+
+---
+
+### Submission flow
+
+1. Accepting a transaction(job)
+
+When a user submits a transaction (e.g. via `SubmitPayForBlobToQueue`):
+
+- Tx is wrapped in a job containing:
+  - transaction data
+  - user options
+  - context
+  - a result channel
+- The queue enforces memory based backpressure:
+  - If sufficient memory is available, the job is appended to the queue
+  - If queue exceeds its memory limit, submission blocks until memory is freed up
+ User should be able to set desired queue size during client initialization.
+- Nothing gets signed yet.
+- No network operations are performed.
+- User will receive tx result asynchronously via the result channel.
```

Comment:
As an alternative implementation we can have only one channel per signer that the broadcaster listens, but not sure if that would be easier or not.

Reply:
this queue is initialised per signer since all txs from the same signer need to be in order. if we had per signer channel how would you store txs?

Reply:
I'm not sure I understand this fully but I don't want to have design suggestions go unnoticed. If you think this would be a better design, would you mind elaborating a little further?